### PR TITLE
da: add submit, get context timeouts

### LINKED ISSUE #1036: DA layer submission failure due to tx timeout
While sending transactions sequentially to a Rollkit sequencing node running Polaris and posting data to a local celestia devnet instance, I encountered an issue which Rollkit reports as:

```
DA layer submission failed error="Post \"http://localhost:26659/submit_pfb\": EOF" attempt=1 module=BlockManager
5:29PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=2 module=BlockManager
5:29PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=3 module=BlockManager
5:29PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=4 module=BlockManager
5:29PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=5 module=BlockManager
5:29PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=6 module=BlockManager
5:29PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=7 module=BlockManager
5:30PM ERR DA layer submission failed error="Codespace: 'sdk', Code: 19, Message: " attempt=8 module=BlockManage
```

Eventually after 30 or so attempts, Rollkit uses a pending block to progress the chain, but doing so causes a height mismatch between the client (Rollkit) and the app (Polaris), which can lead to a panic.

Looking at the logs for the local celestia devnet shows that the first failed submission occurs because of `ERR Error on broadcastTxCommit err="timed out waiting for tx to be included in a block" module=rpc`, while any future attempt by Rollkit to submit the data fails because `ERR Error on broadcastTxCommit err="tx already exists in cache" module=rpc`. 

While it might be possible to change the `timeout_commit` value locally and resolve the issue, Rollkit might be better off using something like `BroadcastTxSync` or `BrodcastTxAsync`, or perhaps the change from `go-cnc` to `open-rpc` will fix this issue by allowing Rollkit to wait for the context deadline of the request more accurately, avoiding further attempts to submit a tx already in cache

### PR DESCRIPTION
**Pull request by tuxcanfly**
## Overview

This PR adds context with sane default timeouts to submit and get blob methods on the DA client. Fixes #1036

## Checklist

- [x] New and updated code has appropriate documentation
- [x] New and updated code has new and/or updated testing
- [x] Required CI checks are passing
- [x] Visual proof for any user facing features like CLI or documentation updates
- [x] Linked issues closed with keywords

## Summary by CodeRabbit

- **New Features**
	- Introduced timeouts for block submission and retrieval to enhance performance and reliability.

- **Bug Fixes**
	- Improved error handling in test scenarios for timeouts and maximum blob size constraints.

- **Tests**
	- Added new tests to cover context cancellation and specific error scenarios.

### CODE REVIEW

File: da/da.go [hunk 1]
```diff
@@ -14,6 +15,14 @@ import (
 	pb "github.com/rollkit/rollkit/types/pb/rollkit"
 )
 
+const (
+	// submitTimeout is the timeout for block submission
+	submitTimeout = 60 * time.Second
```
Comment (Manav-Aggarwal) [inline da/da.go:20]: Is this something we're fine with hardcoding regardless of the DA layer? Do you think we should make this configurable for the future? Thoughts? @gupadhyaya

Comment (gupadhyaya) [reply da/da.go:20]: @tuxcanfly 
* any reason for choosing 60s not 30s? 
* do you need two separate timeouts for submit and retrieve? 
* @Manav-Aggarwal i think we can use set defaults for now and later make it configurable when we have multiple DA support. this is still an safe upper-bound.

File: da/da.go [hunk 2]
```diff
@@ -111,7 +120,9 @@ func (dac *DAClient) SubmitBlocks(ctx context.Context, blocks []*types.Block) Re
 			},
 		}
 	}
+	ctx, cancel := context.WithTimeout(ctx, submitTimeout)
 	ids, _, err := dac.DA.Submit(ctx, blobs, dac.GasPrice)
+	cancel()
```
Comment (gupadhyaya) [inline da/da.go:125]: `defer cancel()`?

File: da/da.go [hunk 3]
```diff
@@ -161,7 +172,9 @@ func (dac *DAClient) RetrieveBlocks(ctx context.Context, dataLayerHeight uint64)
 		}
 	}
 
+	ctx, cancel := context.WithTimeout(ctx, retrieveTimeout)
 	blobs, err := dac.DA.Get(ctx, ids)
+	cancel()
```
Comment (gupadhyaya) [inline da/da.go:177]: `defer cancel()`?

File: da/da.go (hunk 1 above)
Comment (tuxcanfly) [reply da/da.go:20]: > any reason for choosing 60s not 30s?

in extreme cases e.g. inscriptions flood, the core endpoint might timeout accepting transactions, it's a sync method right now (`BroadcastTx`).

> do you need two separate timeouts for submit and retrieve?

Yes as both have different contexts and implementations (e.g. light nodes might require more time than bridge nodes).

> Is this something we're fine with hardcoding regardless of the DA layer? 

agreed, defaults for now but should be configurable.

File: da/da.go (hunk 2 above)
Comment (tuxcanfly) [reply da/da.go]: Done.

File: da/da.go (hunk 3 above)
Comment (tuxcanfly) [reply da/da.go]: Done.

**Review by Manav-Aggarwal — APPROVED** [review]

**Comment by gupadhyaya** [comment]
@tuxcanfly is there a way to test this? i don't think it is addressing any problem.

**Review by MSevey — APPROVED** [review]

**Comment by tuxcanfly** [comment]
@gupadhyaya added test case for context timeout. The submit call relies on a sync endpoint in core and can stall the block submission loop if we don't use a context deadline, see: https://gist.github.com/tuxcanfly/207ab3dc1fac4bb9c38bedad2a053f18

**Review by Manav-Aggarwal — APPROVED** [review]

**Review by gupadhyaya — APPROVED** [review]

**Pull request merged**